includes/api/class-wc-admin-rest-leaderboards-controller.php

```diff
@@ -49,18 +49,251 @@ public function register_routes() {
 		);
 	}
 
+	/**
+	 * Get the data for the coupons leaderboard.
+	 *
+	 * @param int    $per_page Number of rows.
+	 * @param string $after Items after date.
+	 * @param string $before Items before date.
+	 */
+	public function get_coupons_leaderboard( $per_page, $after, $before ) {
+		$coupons_data_store = new WC_Admin_Reports_Coupons_Data_Store();
+		$coupons_data       = $coupons_data_store->get_data(
+			array(
+				'orderby'       => 'orders_count',
+				'order'         => 'desc',
+				'after'         => $after,
+				'before'        => $before,
+				'per_page'      => $per_page,
+				'extended_info' => true,
+			)
+		);
+
+		$rows = array();
+		foreach ( $coupons_data->data as $coupon ) {
+			$coupon_url  = wc_admin_url( 'analytics/coupons?filter=advanced&coupon_includes=' . $coupon['coupon_id'] );
+			$coupon_code = $coupon['extended_info'] && $coupon['extended_info']['code'] ? $coupon['extended_info']['code'] : '';
```

Comment:
This looks a little too JavaScripty - won't this throw warnings if `extended_info` or `code` aren't present in the array? I think we'll need to use `isset()` here.

Reply:
Good call 👍

Reply:
> This looks a little too JavaScripty 

never. you can never be too JavaScripty.